algorithms/src/snark/marlin/ahp/indexer/constraint_system.rs

```diff
@@ -124,6 +150,21 @@ impl<F: Field> CS<F> for ConstraintSystem<F> {
         self.c.push(Self::make_row(&c(LinearCombination::zero())));
 
         self.num_constraints += 1;
+        self.s_mul.push(F::one());
+        self.s_lookup.push(F::zero());
+    }
+
+    fn lookup(&mut self, val: LinearCombination<F>) -> Result<Variable, SynthesisError> {
```

Comment:
This is a unary lookup, right? I think this suffices for Sinsemillia, but we should also support binary lookups `lookup(a, b) = c`, since those are useful for things like XOR, ANDs, etc.